pkg/client/restclient/request.go

```diff
@@ -805,10 +805,12 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 		}
 		resp, err := client.Do(req)
 		updateURLMetrics(r, resp, err)
-		if err != nil {
-			r.backoffMgr.UpdateBackoff(r.URL(), err, 0)
-		} else {
-			r.backoffMgr.UpdateBackoff(r.URL(), err, resp.StatusCode)
+		if r.baseURL != nil {
```

Comment:
If the point is not to call the below when r.URL() is "", then please check `r.URL() != ""` directly rather than making readers understand the relation between baseURL and URL().

However it's not clear to me why it matters? Why should going to `/` be exempted from the backoff?